docs/upgrading_renku.rst

```diff
@@ -0,0 +1,69 @@
+.. upgrading_renku:
+
+Upgrading a Renku Project
+=========================
+
+The Renku Project is in constant flux! This means that often there are changes
+to RenkuLab (e.g. the web platform (UI) and the knowledge graph (KG)), and ``renku``
+(the command-line interface (CLI), which is installed via ``pipx`` in the
+interactive environments).
+
+Using new features on RenkuLab
+------------------------------
+
+New features in the UI and KG on RenkuLab are announced in the :ref:`release_notes`.
+
+When we update components of the RenkuLab platform, usually you will have access
+to these new features without doing anything special. E.g. when we added the ability
+for anonymous (i.e. not logged in) users to launch interactive environments, all
+projects were immediately accessible to anonymous users without the project owner
+needing to change any settings.
+
+.. _renku_cli_upgrade:
+
+Using new ``renku`` CLI features
+--------------------------------
+
+
+.. note::
+
+  This section is for upgrading the version of ``renku`` CLI installed into
+  the Interactive Environments on RenkuLab for your project. See :ref:`upgrading_local`
+  for upgrading your local machine's version of ``renku``.
+
+When we release a new version of the ``renku`` CLI, you do have to make some
+(minimal) changes to the ``Dockerfile`` in your project to ensure that the
+Interactive Environment on RenkuLab will use the image with the correct version.
+
+The version of the ``renku`` CLI is defined in the base image specified in the
+``FROM`` line (usually line 1) of the ``Dockerfile`` in your repo.
+
+1. Replace this line with the latest available image for the "flavor" you're using,
+   which you can find by:
+
+* checking the `renkulab-docker repo README.md <https://github.com/SwissDataScienceCenter/renkulab-docker/blob/master/README.md>`_
+  for the naming conventions of the "flavor" of image you're updating (e.g. if you're using
+  the "minimal python project", you will look in the ``py`` section)
+* visiting the dockerhub page linked in the naming conventions section in the above ``README``
+* choosing the latest available tagged image
+
+Note that the version of the ``renku`` CLI is defined in the image name. For example,
+``renku/renkulab-py:3.7-renku0.10.3-0.6.2`` has CLI version ``0.10.3``. Please choose
+an image with the renku version specified (e.g. ``-renku0.10.3-`` and not ``-renku-``),
+unless you absolutely need the latest development version of ``renku``, since it makes
+debugging and reproducibility much simpler.
+
+2. Push your changes (either from the GitLab editor, or in a running Interactive
+   Environment), and start up a new Interactive Environment to use this new image.
+
+3. When you start to run ``renku`` commands, you might be asked to call ``renku migrate``.
+   This command ensures that the metadata in your project is updated.
+
+.. note::
+
+  ``renku`` is installed in the image with ``pipx``. You can also upgrade ``renku``
```

Comment:
This section on upgrading the renku version inside a running environment using pipx is true, but is it necessary to inform our users about it? I think we should either drop it, or explain why you might want to do this (for example, testing out a new version of renku to see if it works with your project).

Reply:
it's also useful to do if you don't want to rebuild your whole image (or can't because of some dependency problem)

Reply:
👍 i turned the note into a section that explains this